### PR TITLE
Allow cloudbackup schedule update

### PR DESCRIPTION
**Pull request by talakad**
Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:
Allows user to update existing cloudsnap with credential or schedule without having to recreate it.
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -572,6 +574,11 @@ type CloudBackupGroupSchedCreateRequest struct {
 	MaxBackups uint
 	// Full indicates if scheduled backups must be full always
 	Full bool
+	// RetentionDays is the number of days that the scheduled backups will be kept
+	// and after these number of days it will be deleted
+	RetentionDays uint32
+	// Update indicates if existing schedule is being updated
+	Update bool
```
Comment (disrani-px) [inline api/api.go]: We should create new APIs for update instead of overloading the Create APIs

Comment (talakad) [reply api/api.go]: OK. Added new API for update.

File: api/api.proto [hunk 2]
```diff
@@ -1864,6 +1864,15 @@ service OpenStorageCloudBackup {
       };
     }
 
+  // Update existing cloud backup schedule
+  rpc SchedUpdate(SdkCloudBackupSchedUpdateRequest)
+    returns (SdkCloudBackupSchedUpdateResponse) {
+      option(google.api.http) = {
+        post: "/v1/cloudbackups/schedules"
```
Comment (disrani-px) [inline api/api.proto]: `post` is for create operations. `put` should be used for updates.

Comment (talakad) [reply api/api.proto]: I missed it. Thanks! Done

**Pull request merged**